shared/backupUtils/restoreData.ts

```diff
@@ -0,0 +1,275 @@
+    await MMKV.setItem(MY_VCS_STORE_KEY, finalVC);
+  }
+
+  /**
+   * Handle cleanup of previous backup state
+   * @param completeBackupData Entire Backup data to be loaded
+   * @param encryptionKey Encryption key
+   * @returns Previous
+   */
+  async function loadVCs(
+    completeBackupData: Record<string, any>,
+    encryptionKey: string,
+  ): Promise<Record<string, any> | undefined> {
+    try {
+      const {VC_Records: allVCs, dataFromDB: dataFromDB} = completeBackupData;
+
+      await processAndStoreVCs(allVCs, dataFromDB, encryptionKey);
+      await updateMyVcList(dataFromDB, encryptionKey);
+      await fileStorage.removeItemIfExist(`${DocumentDirectoryPath}/.prev`);
+      return dataFromDB;
+    } catch (error) {
+      console.error('Error while loading VCs:', error);
+      sendErrorEvent(
+        getErrorEventData(
+          TelemetryConstants.FlowType.dataBackup,
+          TelemetryConstants.ErrorId.failure,
+          error.message,
+        ),
+      );
+    }
+  }
+
+  /**
+   * Process and store individual Verifiable Credentials
+   * @param allVCs All Verifiable Credentials from backup
+   * @param dataFromDB Database data containing VC metadata
+   * @param encryptionKey Encryption key
+   */
+  async function processAndStoreVCs(
+    allVCs: Record<string, any>,
+    dataFromDB: Record<string, any>,
+    encryptionKey: string,
+  ) {
+    const vcKeys = Object.keys(allVCs);
+
+    await Promise.all(
+      vcKeys.map(async key => {
+        const vcData = allVCs[key];
+        // Add randomness to timestamp to maintain uniqueness
+        const timestamp =
+          Date.now() + Math.random().toString().substring(2, 10);
+        const prevUnixTimeStamp = vcData.vcMetadata.timestamp;
+
+        // Verify and update the credential
+        const isVerified = await verifyCredential(
+          vcData,
+          vcData.vcMetadata,
+        );
+        vcData.vcMetadata.timestamp = timestamp;
+        vcData.vcMetadata.isVerified = isVerified;
+
+        //update the vcMetadata
+        dataFromDB.myVCs.forEach(myVcMetadata => {
+          if (
+            myVcMetadata.requestId === vcData.vcMetadata.requestId &&
+            myVcMetadata.timestamp === prevUnixTimeStamp
+          ) {
+            myVcMetadata.timestamp = timestamp;
+            myVcMetadata.isVerified = isVerified;
+          }
+        });
+
+        // Encrypt and store the VC
+        const updatedVcKey = new VCMetadata(vcData.vcMetadata).getVcKey();
+        const encryptedVC = await encryptJson(
+          encryptionKey,
+          JSON.stringify(vcData),
+        );
+        await Storage.setItem(updatedVcKey, encryptedVC, encryptionKey);
+      }),
+    );
+  }
+
+  /**
+   * Verify the restored credential
+   * @param vcData  Verifiable Credential from backup
+   * @param vcMetadata  VC metadata
+   */
+  async function verifyCredential(vcData: any, vcMetadata: any) {
```

Comment:
verifyCredential can we keep this function in util and use in all flows like download credential, restore

Reply:
Done